openassessment/workflow/models.py

```diff
@@ -353,11 +422,27 @@ def set_score(self, score):
                 'points_possible'.
 
         """
-        sub_api.set_score(
-            self.submission_uuid,
-            score["points_earned"],
-            score["points_possible"]
-        )
+        if not self.staff_score_exists():
```

Comment:
We're currently in the workflow model, this is how we ensure that no other scores can be recorded after a staff score has been recorded.